cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -2157,11 +2158,19 @@ private Range getNamespaceRange(String namespaceId) {
       ImmutableList.of(Fields.stringField(StoreDefinition.AppMetadataStore.NAMESPACE_FIELD, namespaceId)));
   }
 
-  private Range getNamespaceAndApplicationRange(String namespaceId, String applicationId) {
+  private Range getNamespaceAndApplicationRange(ApplicationReference appReference) {
     return Range.singleton(
       ImmutableList.of(
-        Fields.stringField(StoreDefinition.AppMetadataStore.NAMESPACE_FIELD, namespaceId),
-        Fields.stringField(StoreDefinition.AppMetadataStore.APPLICATION_FIELD, applicationId)));
+        Fields.stringField(StoreDefinition.AppMetadataStore.NAMESPACE_FIELD, appReference.getNamespace()),
+        Fields.stringField(StoreDefinition.AppMetadataStore.APPLICATION_FIELD, appReference.getApplication())));
+  }
+
+  private Range getLatestApplicationRange(ApplicationReference appReference) {
+    return Range.singleton(
+      ImmutableList.of(
+        Fields.stringField(StoreDefinition.AppMetadataStore.NAMESPACE_FIELD, appReference.getNamespace()),
+        Fields.stringField(StoreDefinition.AppMetadataStore.APPLICATION_FIELD, appReference.getApplication()),
+        Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, "true")));
```

Comment:
this will need to be changed when https://github.com/cdapio/cdap/pull/14708 is merged

Reply:
yes, I'll note that